### ISSUE TITLE
[$250] Request Money - Text in the compose box changes from "Say hello" to "Write something" & vice versa

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#11471]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to URL https://staging.new.expensify.com/
2. Login with any account
3. Click on FAB button/Request money or Split bill
4. Сomplete the action
5. Pay attention to the text in the input field

Also, 

1. Click on FAB button/New chat
2. Select new user with no previous chat
3. Pay attention to compose box
4. Placeholder text changes

## Expected Result:
The text in the input field shouldn't changes

## Actual Result:
The text in the input field changes from "Say hello" to "Write something." from "Say hello" to "Write something."

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2.10.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/193309858-87395bdb-d279-43ef-a283-ed04c9e244f0.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MonilBhavsar** [comment]
Not a critical bug, looking it later

**Comment by MonilBhavsar** [comment]
I think this is more general issue and also occurs when - 

1. Click on FAB button/New chat
2. Select new user with no previous chat
3. Pay attention to compose box
4. Placeholder text changes

**Comment by Uros787** [comment]
### Proposal
The issue here is that while the chat is loading, our fallback value is "Say hello"
Now we have two options:
Either change the fallback text to Write something
Or completely delete the fallback value so it is an empty string and after loading is finished display the proper value

That would be done here.

https://github.com/Expensify/App/blob/6551eb9d07f54c222bd6cb7411b7613f3ccd03b4/src/pages/home/report/ReportActionCompose.js#L244-L258

Like this:
```
// Waiting until ONYX variables are loaded before displaying the input place holder
        if (_.isEmpty(this.props.reportActions)) {
            return '';
        }
```

**Comment by dhairyasenjaliya** [comment]
**Proposal**

**RCA**

- This happens because in our code base we have added validation to check if `isEmptyChat` which basically returns the length of `this.props.reportActions` and if chat is empty we are returning `1` as even empty chat `reportActions` object will be  containing one record of `creating` 

- Text in the input field changes from "Say hello" to "Write something." because when we are updating a `reportActions` at particular moment  `reportActions` length would be `0` empty so we need to add one more validation to check if  `reportActions` is not `0`

```
        if (this.isEmptyChat() && _.size(this.props.reportActions) !== 0) {
            return this.props.translate('reportActionCompose.sayHello');
        }
```

https://github.com/Expensify/App/blob/44bf1ffc491cd1e99258ddbe07823a99b80e9aad/src/pages/home/report/ReportActionCompose.js#L253-L255

**Comment by mananjadhav** [comment]
Went through the proposals and @dhairyasenjaliya's proposal [here](https://github.com/Expensify/App/issues/11471#issuecomment-1271544517) looks good to me.

@MonilBhavsar All yours.

🎀 👀 🎀 
C+ reviewed

**Comment by MonilBhavsar** [comment]
@dhairyasenjaliya could you please check if this issue use case is also fixed- https://github.com/Expensify/App/issues/11471#issuecomment-1271485274

**Comment by dhairyasenjaliya** [comment]
> @dhairyasenjaliya could you please check if this issue use case is also fixed- [#11471 (comment)](https://github.com/Expensify/App/issues/11471#issuecomment-1271485274)

I did check this condition now, so basically, here we are adding a conditional check when `reportAction` is not empty at that time we display `Say Hello`, but if we want to add this condition then we should probably add a condition whereby if `reportAction` is blank at that time we need to return placeholder as empty since only after data is loaded we can identify the type of placeholder (E.g Say Hello/Write something)

**Comment by mananjadhav** [comment]
@MonilBhavsar @dhairyasenjaliya I am not able to reproduce this in a general scenario. Here's the video from staging. Am I missing something on the reproduction steps?

https://user-images.githubusercontent.com/3069065/196001475-20c8b290-6f7e-49fd-9665-8df7203ac3d4.mp4

**Comment by mananjadhav** [comment]
yeah Request a money I am aware but not with New Chat as mentioned in the GH body. Hence, clarifying.

**Comment by mananjadhav** [comment]
(if you check the video), I am trying a very random email that don't exist in the system

**Comment by MonilBhavsar** [comment]
Hmm, I'm not also able to reproduce now.
@kbecciv could you please confirm if this issue is still reproducible. Thank you!

**Comment by kbecciv** [comment]
@MonilBhavsar Checking with team, will update you shortly

**Comment by kbecciv** [comment]
@MonilBhavsar Issue is still reproduced with build 1.2.17.4

https://user-images.githubusercontent.com/93399543/196500038-c8ff1eac-6052-48e9-9849-50921e336efd.mp4

**Comment by slafortune** [comment]
https://www.upwork.com/ab/applicants/1582488161449332736/job-details

**Comment by MonilBhavsar** [comment]
Thanks!
Yes, reproducible with request money flow. Waiting for proposals...

**Comment by dhairyasenjaliya** [comment]
@MonilBhavsar https://github.com/Expensify/App/issues/11471#issuecomment-1271544517 this proposal is for request money flow can you confirm @mananjadhav

**Comment by mananjadhav** [comment]
@MonilBhavsar Just for request flow, I shortlisted a proposal if you check the comment [here](https://github.com/Expensify/App/issues/11471#issuecomment-1273743342)

**Comment by MonilBhavsar** [comment]
Thanks, had a look.
When we request from a completely new account, the transition still happens, but I think that is because of a new reportAction is added at that moment.

**Comment by mananjadhav** [comment]
@MonilBhavsar Any updates on this one?

**Comment by mananjadhav** [comment]
Quick bump on this @MonilBhavsar

**Comment by MonilBhavsar** [comment]
Proposal looks good to me. let's implement it

**Comment by mananjadhav** [comment]
@dhairyasenjaliya This PR is in draft for a few days. Can we please have it closed soon? Can you please send us an update on this one.

**Comment by mananjadhav** [comment]
I will only be able to check this tomorrow. @kbecciv Is this still reproducible? I am guessing this will go through KI today/tomorrow.

**Comment by mananjadhav** [comment]
You're right @dhairyasenjaliya. 

@MonilBhavsar @kbecciv I am not able to reproduce this anymore on staging. Can you confirm once? I think we need to hold this, until 1-2 KI tests.

**Comment by kbecciv** [comment]
Checking now, update you shortly

**Comment by kbecciv** [comment]
Issue is still reproducible

https://user-images.githubusercontent.com/93399543/200344779-9c7026ab-1473-441a-afcb-ddb0082d8a7c.mp4

.

**Comment by mananjadhav** [comment]
Okay thanks for the comment @dhairyasenjaliya.

Based on the previous comment, this is still open for proposals. @slafortune @MonilBhavsar can we please add `Help Wanted` tag?

**Comment by MonilBhavsar** [comment]
Yes 👍 
@dhairyasenjaliya feel free to propose the solution again

**Comment by mananjadhav** [comment]
Waiting for proposals.

**Comment by MonilBhavsar** [comment]
I tried reproducing the issue and looks like it is reproducible with account having higher number of chats and messages.

**Comment by slafortune** [comment]
Job posted - https://www.upwork.com/ab/applicants/1592579987018907648/job-details

**Comment by jasperhuangg** [comment]
Hey @MonilBhavsar can you maybe provide the login credentials of the account? that makes sense because those accounts would have reports that take longer to load.

I can take this one on, have a solution in mind. I feel like this exact situation is what [`isLoadingReportActions` is for](https://github.com/Expensify/App/blob/7de9ab2d62849fc670879a680dd46cf8003697f8/src/libs/actions/Report.js#L509-L510).

**Comment by marcaaron** [comment]
Not able to repro this on staging or production.

**Comment by jasperhuangg** [comment]
@marcaaron are you using accounts with very large reports?

**Comment by marcaaron** [comment]
What does "very large reports" mean?

**Comment by jasperhuangg** [comment]
Actually now that you mention it, I'm not really sure what they were talking about since we paginate loading report actions. But I'm getting this from [here](https://github.com/Expensify/App/issues/11471#issuecomment-1313313332)

**Comment by marcaaron** [comment]
I think maybe whatever was causing it is not happening anymore? Can you try to reproduce on staging and production to validate what I'm seeing?

**Comment by jasperhuangg** [comment]
cc @MonilBhavsar, since I don't have access to those test accounts can you try to repro again? Since the last time you tried a number of deploys have went out, so perhaps something has changed.

**Comment by MonilBhavsar** [comment]
> Hey @MonilBhavsar can you maybe provide the login credentials of the account? that makes sense because those accounts would have reports that take longer to load.

TBH I just used my original expensify.com account and encountered this. I used my expensifail.com account (less chats) and didn't see the bug. So that led me to this belief.

**Comment by MonilBhavsar** [comment]
I tested right now and was able to reproduce on my expensify account.

**Comment by mananjadhav** [comment]
I think it must be impacting high traffic account, because in the following block we check:

https://github.com/Expensify/App/blob/3528ad556579f3048eb91aa5ec02a2e98b35229b/src/pages/home/report/ReportActionCompose.js#L255-L261

which consists of the check:
```
return _.size(this.props.reportActions) === 1;
```

`reportActions` is passed all the way from `ReportScreen`. Hence while the `reportActions` aren't loaded it shows `Say Hello`.


Question from first principles: why do we have two placeholders? Any specific need for the same between an empty/existing chat?

**Comment by slafortune** [comment]
not overdue

**Comment by MonilBhavsar** [comment]
I investigated this issue today, Root cause seems like - When we create IOU report and navigate to a report (chat), we have an IOU report as a prop in `ReportScreen` component and thus `isLoadingInitialReportActions` becomes false and we render the `ReportFooter` as well as `ReportActionCompose`. In fact it is wrong, we haven't loaded initial report actions, we just have one IOU report as a reportAction and we render the screens.

I believe these are steps to surely reproduce the issue - 

1. Signout (So Onyx is cleared and we'll be required to fetch reportActions)
2. Have two chats A and B with some good number of chats 
3. Navigate to chat A
4. From `+` option, request a Money from Chat B
6. Keep an eye on compose box and see the text changing

I'l revisit again...

> why do we have two placeholders? Any specific need for the same between an empty/existing chat?
@mananjadhav sorry I don't know the reason behind it.

**Comment by MonilBhavsar** [comment]
PR is in review to have only placeholder
And new bug reported to create a new followup issue to fix root cause

**Comment by mananjadhav** [comment]
@MonilBhavsar This doesn’t need C+ review ?

**Comment by MonilBhavsar** [comment]
Sorry forgot, requested your review.

**Comment by MonilBhavsar** [comment]
PR is merged and followup issue to fix the root cause is here https://github.com/Expensify/App/issues/13074

**Comment by slafortune** [comment]
Offer sent to @mananjadhav 👍

**Comment by MonilBhavsar** [comment]
PR is deployed to production and follow up issue is created
@slafortune I think we can close this now?

**Comment by MonilBhavsar** [comment]
@slafortune can we please issue the payment to C+ here

**Comment by slafortune** [comment]
Paid!
